### PR TITLE
i18n(hi): update outdated hindi pages

### PR DESCRIPTION
**Pull request by at-the-vr**
#### Description
- updates [components](https://starlight.astro.build/hi/guides/components/), [css-and-tailwind](https://starlight.astro.build/hi/guides/css-and-tailwind/), [customization](https://starlight.astro.build/hi/guides/customization/),  [i18n](https://starlight.astro.build/hi/guides/i18n/) & [site-search](https://starlight.astro.build/hi/guides/site-search/) in Hindi Language

### CODE REVIEW

**Comment by at-the-vr** [comment]
I am confused with the build error, its pointing at `css-and-tailwind.html` where `Steps` is missing child components. What is the intended fix here ?

**Comment by HiDeoo** [comment]
> I am confused with the build error, its pointing at `css-and-tailwind.html` where `Steps` is missing child components. What is the intended fix here ?

The error is actually pointing to `docs/src/content/docs/hi/guides/customization.mdx` and the issue is the first closing `</Steps>` in this page which is at the moment before "### हल्के और गहरे रंग के लोगो वेरिएंट" but it should just be before "डिफ़ॉल्ट रूप से, लोगो आपकी साइट के `title` के साथ प्रदर्शित होगा।" (basically ~15 ish lines above its current position)

**Comment by at-the-vr** [comment]
thanks Hideoo 🙌

File: docs/src/content/docs/hi/guides/components.mdx [hunk 1]
```diff
@@ -296,3 +278,118 @@ import importedCode from '/src/env.d.ts?raw';
...
+
+1. Import the component into your MDX file:
+
+   ```js
+   import { Steps } from '@astrojs/starlight/components';
+   ```
+
+2. Wrap `<Steps>` around your ordered list items.
+
+</Steps>
```
Comment (delucis) [inline docs/src/content/docs/hi/guides/components.mdx:368]: Up to you @at-the-vr, but the language in the example could also be translated to Hindi here.

**Review by delucis — COMMENTED** [review]
Looks great @at-the-vr! Just left one small suggestion.

File: docs/src/content/docs/hi/guides/components.mdx (hunk 1 above)
Comment (at-the-vr) [reply docs/src/content/docs/hi/guides/components.mdx]: Wow can't believe I omitted that part out 🤦 . Resolved with latest push 👍

**Review by delucis — APPROVED** [review]
Perfect, thanks for that last detail @at-the-vr 🙌

**Pull request merged**